### PR TITLE
Add: Issue template forms for users to create better bug and documentation reports

### LINKED ISSUE #408: Create issue templates for issues :)
Hi Friends! 👋 

it would be helpful for our users to have an issue template(s) that people can use to report bugs, request new features, (anything else). 

I will go ahead and work on this looking to other repos for examples of nice templates. if you have any suggestions please add in this issue!!

### PR DESCRIPTION
**Pull request by lwasser**
closes #408

## Description

This is just a way to help guide uses to better bug reports that we can begin to evaluate more easily
you can see what the [templates look like in my repo here!](https://github.com/lwasser/stravalib/issues/new/choose)

Screen Shot 2023-08-16 at 12 08 04 PM

## Type of change

Select the statement best describes this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [x] Other (please describe) this is an infrastructure update

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [x] This change doesn't require tests

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/documentation-issue.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name:  📝 Documentation Improvement :smi
```
Comment (lwasser) [inline .github/ISSUE_TEMPLATE/documentation-issue.yml:1]:
```suggestion
name:  📝 Documentation Improvement
```

**Comment by lwasser** [comment]
ok friends - [have a look here to test these out in my repo](https://github.com/lwasser/stravalib/issues/new/choose) - let me know if you catch any issues / typos or want additional features. i decided to turn these into forms and added a config to direct folks to our discussions for general questions vs bug or documentation reports. CI will likely yell at us for poor test coverage but that doesn't relate to this issue!!

**Review by enadeau — CHANGES_REQUESTED** [review]
I globally really likes the template, only left two small comments.

I'm also thinking whether we should have a template for feature request? Or maybe allow blank issue for issue that don't really fit in one of the templates?

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      description: >
+        Please describe or show a code example of the expected behavior.
+    validations:
+      required: true
+  - type: textarea
+    id: version
+    attributes:
+      label: Your environment
+      description: >
+        Please paste the output of ``pip list`` or ``conda list`` here
```
[the remark this answers is not in the document]
Comment (lwasser) [reply .github/ISSUE_TEMPLATE/bug-report.yml:86]: ok no problem - i can make that change (likely will be in the next few days!)... i'll do the same with conda

`conda env export > environment.yml` for easier reproducibility!

File: .github/ISSUE_TEMPLATE/documentation-issue.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        - label: >
+            I have checked that this issue has not already been reported.
+            required: true
+        - label: >
+            I have confirmed this bug exists on the
+            [latest version](https://pypi.org/project/stravalib/) of stravalib.
+        - label: >
+            I have confirmed this bug exists on the
+            [main branch](https://github.com/stravalib/stravalib)
+            of stravalib.
```
[the remark this answers is not in the document]
Comment (lwasser) [reply .github/ISSUE_TEMPLATE/documentation-issue.yml:20]: oh gosh - that is a mistake.  i'll fix. i must have overwritten something while rebasing or something? 🤷‍♀️ i was using the [pandas templates](https://raw.githubusercontent.com/pandas-dev/pandas/main/.github/ISSUE_TEMPLATE/documentation_improvement.yaml) which are nice. there should only be one box to ensure they are building docs or looking at docs from main.

**Comment by lwasser** [comment]
thank you all for the feedback!! 🚀  i'll make a few changes in the next few days so we can merge these and the nox pr!

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lwasser) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: just pushed a fix for this

File: .github/ISSUE_TEMPLATE/documentation-issue.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lwasser) [reply .github/ISSUE_TEMPLATE/documentation-issue.yml]: this was pushed as well. i haven't merged either fix into my main branch - YET for testing but will

**Comment by lwasser** [comment]
ok friends, [you can view the updates here](https://github.com/lwasser/stravalib/issues/new/choose) 
below is a screenshot. i enabled empty issues just in case as an option. 

Screen Shot 2023-08-20 at 6 45 59 PM

Let me know if you catch anything else!!

**Comment by lwasser** [comment]
Ok ready for review!! 

> Nice work very helpful

thank you @yihong0618 !!

**Comment by lwasser** [comment]
ok great - with two reviews i'm going to merge this. Jonatan is super busy this week but we can always revisit in the future if we need to make changes!! thank you @yihong0618 and @enadeau for the reviews!!

**Pull request merged**